src/messaging/tests/TestReliableMessageProtocol.cpp

```diff
@@ -512,58 +535,44 @@ void CheckResendApplicationMessageWithLostAcks(nlTestSuite * inSuite, void * inC
         1, // CHIP_CONFIG_RMP_DEFAULT_ACTIVE_RETRY_INTERVAL
     });
 
-    mockReceiver.mDropAckResponse = true;
-
-    // Let's not drop any messages. We'll drop acks for this test.
+    // Let's not drop the message. Expectation is that it is received by the peer, but the ack is dropped
     gLoopback.mSendMessageCount    = 0;
     gLoopback.mNumMessagesToDrop   = 0;
     gLoopback.mDroppedMessageCount = 0;
 
+    // Drop the ack, and also close the peer exchange
+    mockReceiver.mDropAckResponse = true;
+    mockReceiver.mRetainExchange  = false;
+
     // Ensure the retransmit table is empty right now
     NL_TEST_ASSERT(inSuite, rm->TestGetCountRetransTable() == 0);
 
     err = exchange->SendMessage(Echo::MsgType::EchoRequest, std::move(buffer));
     NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
-    exchange->Close();
```

Comment:
We should be able to keep closing the exchange here; it should keep itself alive as it resends, right?